### PR TITLE
FUSETOOLS-2843 - renaming to Red Hat Fuse

### PR DESCRIPTION
**Pull request by bfitzpat**
Signed-off-by: Brian Fitzpatrick <bfitzpat@redhat.com>

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [x] Did you use the Jira Issue number in the commit comments?
- [x] Did you set meaningful commit comments on each commit?
- [x] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [ ] Did the CI job report a successful build?
- [ ] Is the non-happy path working, too?

## Maintainability

- [ ] Are all Sonar reported issues fixed or can they be ignored?
- [x] Is there no duplicated code?
- [x] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [x] Have you used the correct file header copyright comment?
- [x] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by bfitzpat** [comment]
@lhein I have done everything but the uitests chunk with Red Deer to ensure that I'm going down the right path on the rename. I believe this is a good start.

**Comment by apupier** [comment]
testPR

**Comment by apupier** [comment]
build is no more available to investigate why it failed, relaunched a build then :(

**Comment by apupier** [comment]
testPR

File: servers/plugins/org.fusesource.ide.server.fuse.core/OSGI-INF/l10n/bundle.properties [hunk 1]
```diff
@@ -8,30 +8,30 @@
 # Contributors:
 #     JBoss by Red Hat - Initial implementation.
 ##############################################################################
-runtimeTypeFuseESBLabel_60=Red Hat JBoss Fuse 6.0
```
Comment (apupier) [inline servers/plugins/org.fusesource.ide.server.fuse.core/OSGI-INF/l10n/bundle.properties]: 6.x ones need to be kept with Red Hat Jboss Fuse, see https://issues.jboss.org/browse/FUSETOOLS-2843?focusedCommentId=13545002&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13545002

File: servers/plugins/org.fusesource.ide.server.fuse.core/OSGI-INF/l10n/bundle.properties [hunk 2]
```diff
@@ -8,30 +8,30 @@
...
+fuseESBServerDescription_63=Server Definition of Red Hat Fuse 6.3
+fuseESBServerType_70=Red Hat Fuse 7.0 Server
+fuseESBServerDescription_70=Server Definition of Red Hat Fuse 7.0
 
-fuseesb.launch.name=Red Hat JBoss Fuse Launcher
+fuseesb.launch.name=Red Hat Fuse Launcher
 redhat=Red Hat JBoss Middleware
 Bundle-Vendor = JBoss by Red Hat
-Bundle-Name = Fuse Server Core Plugin for JBoss Fuse
-runtimeDetector.name = Red Hat JBoss Fuse
```
Comment (apupier) [inline servers/plugins/org.fusesource.ide.server.fuse.core/OSGI-INF/l10n/bundle.properties]: I guess when the same is used for 6.x and 7.x we can migrate the name

File: servers/plugins/org.fusesource.ide.server.fuse.core/plugin.xml [hunk 3]
```diff
@@ -172,7 +172,7 @@
   <extension
         point="org.eclipse.debug.core.launchConfigurationTypes">
      <launchConfigurationType
-           category="Red Hat JBoss Fuse"
+           category="Red Hat Red Hat Fuse"
```
Comment (apupier) [inline servers/plugins/org.fusesource.ide.server.fuse.core/plugin.xml]: bad replacement

File: servers/plugins/org.fusesource.ide.server.fuse.core/plugin.xml [hunk 4]
```diff
@@ -182,7 +182,7 @@
 	   sourcePathComputerId="org.fusesource.ide.server.karaf.core.server.sourcePathComputerDelegate">
      </launchConfigurationType>
      <launchConfigurationType
-           category="Red Hat JBoss Fuse"
+           category="Red Hat Red Hat Fuse"
```
Comment (apupier) [inline servers/plugins/org.fusesource.ide.server.fuse.core/plugin.xml]: bad replacement

File: servers/plugins/org.fusesource.ide.server.fuse.core/src/org/fusesource/ide/server/fuse/core/bean/ServerBeanTypeFuse6x.java [hunk 5]
```diff
@@ -34,7 +34,7 @@
 	
 	protected ServerBeanTypeFuse6x() {
 		super(	"FUSE6x", //$NON-NLS-1$
-				"JBoss Fuse 6.x", //$NON-NLS-1$
+				"Red Hat Fuse 6.x", //$NON-NLS-1$
```
Comment (apupier) [inline servers/plugins/org.fusesource.ide.server.fuse.core/src/org/fusesource/ide/server/fuse/core/bean/ServerBeanTypeFuse6x.java]: 6.x need to be kept as-is https://issues.jboss.org/browse/FUSETOOLS-2843?focusedCommentId=13545002&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13545002

File: servers/tests/org.fusesource.ide.server.tests/src/main/java/org/fusesource/ide/server/tests/locator/FuseESBRuntime6xLocatorIT.java [hunk 6]
```diff
@@ -61,7 +61,7 @@ public void testFuseESB() throws Exception {
 		MockListener listener = new MockListener();
 		locator.searchForRuntimes(dest, listener, new NullProgressMonitor());
 		
-		assertThat(listener.getFoundRuntime().getName()).contains("Red Hat JBoss Fuse 6");
+		assertThat(listener.getFoundRuntime().getName()).contains("Red Hat Fuse 6");
```
Comment (apupier) [inline servers/tests/org.fusesource.ide.server.tests/src/main/java/org/fusesource/ide/server/tests/locator/FuseESBRuntime6xLocatorIT.java]: https://issues.jboss.org/browse/FUSETOOLS-2843?focusedCommentId=13545002&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13545002

File: uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/FeaturesServerTest.java [hunk 7]
```diff
@@ -72,7 +72,7 @@ public void clean() {
 	 */
 	@Test
 	public void testFuseJbossRuntimeDetection() {
-		downloadRuntime("JBoss Fuse 6.3.0");
+		downloadRuntime("Red Hat Fuse 6.3.0");
```
Comment (apupier) [inline uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/FeaturesServerTest.java]: need to be kept as-is for 6.x https://issues.jboss.org/browse/FUSETOOLS-2843?focusedCommentId=13545002&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13545002

File: uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/FeaturesServerTest.java [hunk 8]
```diff
@@ -110,7 +110,7 @@ private void downloadRuntime(String name) {
 		runtimeWiz.setInstallFolder(getPath() + "target/temp_install");
 		runtimeWiz.setDownloadFolder(getPath() + "target/temp");
 		runtimeWiz.finish(name);
-		new WaitWhile(new ShellIsAvailable("Download 'JBoss Fuse 6.3.0'"), TimePeriod.getCustom(900));
+		new WaitWhile(new ShellIsAvailable("Download 'Red Hat Fuse 6.3.0'"), TimePeriod.getCustom(900));
```
Comment (apupier) [inline uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/FeaturesServerTest.java]: need to be kept as-is for 6.x https://issues.jboss.org/browse/FUSETOOLS-2843?focusedCommentId=13545002&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13545002

File: uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/RegressionTest.java [hunk 9]
```diff
@@ -183,7 +183,7 @@ public void issue_1076() {
 		new WaitUntil(new ShellIsAvailable("New Server Runtime Environment"));
 		new DefaultShell("New Server Runtime Environment").setFocus();
 		AbstractWait.sleep(TimePeriod.SHORT);
-		new DefaultTreeItem("JBoss Fuse", "JBoss Fuse 6.1").select();
+		new DefaultTreeItem("Red Hat Fuse", "Red Hat Fuse 6.1").select();
```
Comment (apupier) [inline uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/RegressionTest.java]: need to be kept as-is for 6.x https://issues.jboss.org/browse/FUSETOOLS-2843?focusedCommentId=13545002&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13545002
(but perhaps only the second-level as the first level might be shared)

**Comment by apupier** [comment]
@apodhrad @tsedmik I tried to update the Red Deer tests too but I'm not able to configure to play all of them locally (I know it's possible but don't have time to do it)
Would be nice to test it directly with this PR, is it possible? Or maybe you prefer to handle it after the PR is merged?

**Comment by bfitzpat** [comment]
@apupier Your changes for the older versions look good to me. I can't approve a PR I created, but I give it a +1 anyway. 👍

**Comment by apupier** [comment]
test in failure:

`Expecting empty but was:<[{"severity"=2, "causeLineNumber"=1, "Creation time: "=1521108752965L, "charEnd"=38, "causeColumnEnd"=0, "message"="Project build error: Non-parseable POM /home/jenkins/.m2/repository/io/syndesis/syndesis-rest-parent/1.2.3/syndesis-rest-parent-1.2.3.pom: only whitespace content allowed before start tag and not o (position: START_DOCUMENT seen o... @1:1) ", "causeResourcePath"="/home/jenkins/.m2/repository/io/syndesis/syndesis-rest-parent/1.2.3/syndesis-rest-parent-1.2.3.pom", "resource affected content"="<?xml version="1.0" encoding="UTF-8"?>`

wtf?

**Comment by apupier** [comment]
argh all check passed but require a new rebase to resolve conflicts

image

**Comment by apupier** [comment]
waiting now for a master green build

**Pull request merged**